Code/Core/FileIO/FileStream.cpp

```diff
@@ -234,7 +234,18 @@ bool FileStream::IsOpen() const
         totalBytesRead += bytesReadNow;
     } while ( totalBytesRead < bytesToRead );
 #elif defined( __APPLE__ ) || defined( __LINUX__ )
-    totalBytesRead += static_cast<uint64_t>( read( m_Handle, buffer, bytesToRead ) );
+    const uint64_t chunkSize = 1073741824;
+    do
+    {
+        const size_t bytesReadNow =
+            read( m_Handle, (char *)buffer + totalBytesRead, chunkSize );
```

Comment:
Reads can be requested that are smaller than the available data, but this code will always attempt to read to the end of the file. This will cause incorrect behavior and buffer overruns.

The tests seem to be catching this problem.

I think it would make sens to unify the do/while loops between the two platforms, including the "tryToReadNow" logic that handles this on Windows. I think only the ReadFile vs Read and checking of the return result needs to be ifdef'd.